backend/src/services/teams/teams.repo.js

```diff
@@ -0,0 +1,231 @@
+'use strict';
+
+/**
+ * Teams Repository
+ *
+ * Data access methods for team-related operations.
+ */
+
+class TeamsRepo {
+  constructor(db) {
+    this.db = db;
+  }
+
+  /**
+   * Get all teams in a course.
+   * @param {number} courseId
+   * @returns {Promise<Array>}
+   */
+  async getTeamsInCourse(courseId) {
+    const teams = await this.db.teams.findMany({
+      where: { course_id: courseId },
+      orderBy: { id: 'asc' },
+    });
+    return teams;
+  }
+
+  /**
+   * Get a single team by ID within a course.
+   * @param {number} courseId
+   * @param {number} teamId
+   * @returns {Promise<Object|null>}
+   */
+  async getTeamById(courseId, teamId) {
+    const team = await this.db.teams.findFirst({
+      where: {
+        id: teamId,
+        course_id: courseId,
+      },
+    });
+    return team;
+  }
+
+  /**
+   * Get members of a team (enrollments + user details).
+   * @param {number} courseId
+   * @param {number} teamId
+   * @returns {Promise<Array>}
+   */
+  async getTeamMembers(courseId, teamId) {
+    const members = await this.db.enrollments.findMany({
+      where: {
+        course_id: courseId,
+        team_id: teamId,
+      },
+      include: {
+        users: true,
+      },
+    });
+    return members;
+  }
+
+  /**
+   * Create a new team in a course.
+   * @param {number} courseId
+   * @param {{ name: string, description?: string|null }} data
+   * @returns {Promise<Object>}
+   */
+  async createTeam(courseId, data) {
+    const { name, description = null } = data || {};
+
+    if (!name) {
```

Comment:
[Suggestion]
Trim the name to ensure whitespace only strings are not accepted.

Reply:
Do the same with description if present so that we don't store unnecessary data.